### ISSUE TITLE
Handling System isolation with script loading

### ISSUE THREAD

**Issue by guybedford** [systemjs/systemjs#803]
Because we have no handle on the before load event for script loading, `System` isolation is unfortunately not compatible with this loading mechanism.

One way to achieve this might be to register loader instances on the System constructor itself, and then have communication flow between instances so that whoever's `System.register` gets called it will propagate to the right one.

The problem is that we might want `System` to be another global entirely (and it will be!).

So the question then is how to deal with this conflict.

Current Proposal: **Explicitly have users chooses between CSP or multiple loaders. For a loader to support CSP it just does need to be the primary `System` global in the page. Any additional loaders that are not the `System` global should not use CSP.**

Currently there is no way to _disable_ CSP, we should document and explain this workflow. Perhaps just supporting:

```javascript
System.config({
  meta: {
    '*': {
      scriptLoad: false
    }
  }
});
```

//cc @drieks this may affect your use case if using multiple loaders (I'm not sure how common it would be) in some places unless we create support for such a mechanism as shown above.

**Comment by drieks** [comment]
We are running two JavaScript Applications in the same Page, both having complete different loader and dependency setups. I think that this is a useful use case, maybe not the "mainstream" approach but it should be possible to setup. Even the loaders can have different versions in our use case. But generally supporting multiple instances will also solve this issue.

**Comment by guybedford** [comment]
@drieks as SystemJS is at the moment, it should now fully support isolation. If you are still getting any issues at all please let me know. This issue specifically is an edge case that should be very rare and would only apply to loading code that is set via meta to be known as `amd`, `register` or `global` code with exports defined.

**Comment by guybedford** [comment]
Implemented in https://github.com/systemjs/systemjs/commit/0dd99468dea0b8fec1ccd1afdc01a58a11ea084c.

**Comment by guybedford** [comment]
Released in SystemJS 0.19.5.
